runtime-common/stdlib/server/url-functions.cpp

```diff
@@ -471,3 +496,38 @@ mixed f$parse_url(const string &s, int64_t component) noexcept {
 
   return url_as_array;
 }
+
+void parse_str_set_value(mixed &arr, const string &key, const string &value) noexcept {
+  const char *key_c = key.c_str();
+  const char *left_br_pos = static_cast<const char *>(std::memchr(key_c, '[', key.size()));
+  if (left_br_pos != nullptr) {
+    return parse_str_set_array_value(arr[string(key_c, static_cast<string::size_type>(left_br_pos - key_c))], left_br_pos,
+                                     static_cast<int>(key_c + key.size() - left_br_pos), value);
+  }
+
+  arr.set_value(key, value);
+}
+
+void f$parse_str(const string &str, mixed &arr) noexcept {
+  if (f$trim(str).empty()) {
+    arr = array<mixed>();
+    return;
+  }
+
+  arr = array<mixed>();
```

Comment:
`arr = array<mixed>();` unconditionally

Reply:
It's a code from `runtime`. That functionality is just moved into `runtime-common`, I don't want to change it now.